src/gui/activityjob.cpp

```diff
@@ -0,0 +1,85 @@
+#include <networkjobs.h>
+#include <qurlquery.h>
+
+#include "activityjob.h"
+
+Q_LOGGING_CATEGORY(lcActivityJob, "nextcloud.gui.activityjob", QtInfoMsg)
+
+namespace OCC {
+
+ActivityJob::ActivityJob(QObject *parent)
+    : QObject(parent)
+{
+}
+
+OcsActivityJob::OcsActivityJob(AccountPtr account, QObject *parent)
+    : ActivityJob(parent)
+    , _account(account)
+{
+}
+
+void OcsActivityJob::queryActivities(const Optional<QString> &objectType, const Optional<QString> &objectId, int limit)
+{
+    _limit = limit;
+    _objectId = objectId;
+    _objectType = objectType;
+    startJsonApiJob();
+}
+
+void OcsActivityJob::startJsonApiJob(const Optional<QString> since)
+{
+    const auto url = _objectType || _objectId ? QLatin1String("ocs/v2.php/apps/activity/api/v2/activity/filter")
+                                              : QLatin1String("ocs/v2.php/apps/activity/api/v2/activity");
+    auto job = new JsonApiJob(_account, url, this);
+    QObject::connect(job, &JsonApiJob::jsonReceived,
+        this, [this, job](const QJsonDocument &json, int statusCode) {
+            jsonApiJobFinished(*job, json, statusCode);
+        });
+
+    QUrlQuery params;
+    if (_objectType) {
+        params.addQueryItem(QLatin1String("object_type"), *_objectType);
+    }
+    if (_objectId) {
+        params.addQueryItem(QLatin1String("object_id"), *_objectId);
+    }
+    if (since) {
+        params.addQueryItem(QLatin1String("since"), *since);
+    }
+    params.addQueryItem(QLatin1String("limit"), QString::number(_limit));
+    job->addQueryParams(params);
+    job->start();
+}
+
+void OcsActivityJob::processActivities(const QJsonDocument &json)
+{
+    qCInfo(lcActivityJob) << "Process activities:" << json.object();
+}
+
+void OcsActivityJob::jsonApiJobFinished(const JsonApiJob &job, const QJsonDocument &json, int statusCode)
+{
+    if (statusCode != 200) {
+        emit error();
+        return;
+    }
+    processActivities(json);
+    const auto reply = job.reply();
+    if (reply->hasRawHeader("Link")) {
+        const auto linkRawHeader = reply->rawHeader("Link");
+        if (linkRawHeader.size() <= 14) {
```

Comment:
I do not know where this `14` constant value is coming from but use a named constant instead of this